packages/talos_agent/src/api.rs

```diff
@@ -0,0 +1,121 @@
+    pub writeset: Vec<String>,
+}
+
+/// The data input from client to agent
+#[derive(Clone, Debug)]
+pub struct CertificationRequest {
+    pub message_key: String,
+    pub candidate: CandidateData,
+    pub created_at: Instant,
+}
+
+/// The data output from agent to client
+#[derive(Clone, Debug)]
+pub struct CertificationResponse {
+    pub xid: String,
+    pub is_accepted: bool,
+}
+
+#[derive(Clone)]
+pub struct AgentConfig {
+    // must be unique for each instance
+    pub agent_id: String,
+    pub agent_name: String,
+    pub cohort_name: String,
+}
+
+/// Kafka-related configuration
+#[derive(Clone)]
+pub struct KafkaConfig {
+    pub brokers: String,
+    // Must be unique for each agent instance. Can be the same as AgentConfig.agent_id
+    pub group_id: String,
+    pub certification_topic: String,
+    pub fetch_wait_max_ms: u64,
+    // The maximum time librdkafka may use to deliver a message (including retries)
+    pub message_timeout_ms: u64,
+    // Controls how long to wait until message is successfully placed on the librdkafka producer queue  (including retries).
+    pub enqueue_timeout_ms: u64,
+    // Group session keepalive heartbeat interval
+    // pub heartbeat_interval_ms: u64,
+    pub log_level: RDKafkaLogLevel,
+}
+
+/// The agent interface exposed to the client
+#[async_trait]
+pub trait TalosAgent {
+    async fn certify(&self, request: CertificationRequest) -> Result<CertificationResponse, String>;
+}
+
+pub type TalosAgentType = dyn TalosAgent + Send + Sync;
+
+#[derive(Clone)]
+pub enum TalosIntegrationType {
+    /// The agent will publish certification requests to kafka
+    Kafka,
+    /// The agent will work in offline mode, simulating decisions
+    InMemory,
+}
+
+/// Builds the agent instance, my default agent will be build using in-memory integration type.
+/// The integration type can be changed via "with_kafka" setting.
+pub struct TalosAgentBuilder {
+    config: AgentConfig,
+    kafka_config: Option<KafkaConfig>,
+    integration_type: TalosIntegrationType,
+}
+
+impl TalosAgentBuilder {
+    pub fn new(config: AgentConfig) -> TalosAgentBuilder {
+        Self {
+            config,
+            kafka_config: None,
+            integration_type: InMemory,
+        }
+    }
+
+    /// When agent is built it will be connected to external kafka broker.
+    pub fn with_kafka(&mut self, config: KafkaConfig) -> &mut Self {
+        self.kafka_config = Some(config);
+        self.integration_type = Kafka;
+        self
+    }
+
+    pub fn build(&self) -> Result<Box<TalosAgentType>, String> {
+        let publisher: Box<PublisherType> = match self.integration_type {
+            Kafka => {
+                let config = &self.kafka_config.clone().expect("Kafka configuration is required");
+                let kafka_publisher = KafkaPublisher::new(self.config.agent_id.clone(), config);
+                Box::new(kafka_publisher)
+            }
+            _ => Box::new(MockPublisher {}),
```

Comment:
Same comment as the `consumer`. A particular implementation for Mock is tightly coupled here. We should make it flexible to pass different Mocks to test different scenarios.

Reply:
I agree, currently it is just a placeholder really. I will change this part.